src/base-button/base-button.style.ts

```diff
@@ -0,0 +1,195 @@
+    );
+  }
+
+  :host([size="medium"]) .button {
+    padding: 0 var(--tap-button-md-padding, var(--tap-sys-spacing-6));
+    height: var(--tap-button-md-height, var(--tap-sys-spacing-10));
+    line-height: var(
+      --tap-button-typography-label-md-height,
+      var(--tap-sys-typography-label-sm-height)
+    );
+    font-size: var(
+      --tap-button-typography-label-md-size,
+      var(--tap-sys-typography-label-sm-size)
+    );
+    font-weight: var(
+      --tap-button-typography-label-md-weight,
+      var(--tap-sys-typography-label-sm-weight)
+    );
+  }
+
+  :host([size="large"]) .button {
+    padding: 0 var(--tap-button-lg-padding, var(--tap-sys-spacing-8));
+    /* FIXME: height of the large button is 52px but we dont have 52px in our tokens */
+    height: var(--tap-button-lg-height, var(--tap-sys-spacing-11));
+    line-height: var(
+      --tap-button-typography-label-lg-height,
+      var(--tap-sys-typography-label-lg-height)
+    );
+    font-size: var(
+      --tap-button-typography-label-lg-size,
+      var(--tap-sys-typography-label-lg-size)
+    );
+    font-weight: var(
+      --tap-button-typography-label-lg-weight,
+      var(--tap-sys-typography-label-lg-weight)
+    );
+  }
+
+  :host([variant="primary"]) .button {
+    background-color: var(
+      --tap-button-color-surface-inverse-primary,
+      var(--tap-sys-color-surface-inverse-primary)
+    );
+    color: var(
+      --tap-button-color-content-on-inverse,
+      var(--tap-sys-color-content-on-inverse)
+    );
+  }
+
+  :host([variant="ghost"]) .button {
+    background-color: var(
+      --tap-button-color-surface-tertiary,
+      var(--tap-sys-color-surface-tertiary)
+    );
+    color: var(
+      --tap-button-color-content-primary,
+      var(--tap-sys-color-content-primary)
+    );
+  }
+
+  :host([variant="naked"]) .button {
+    background-color: transparent;
+    color: var(
+      --tap-button-color-content-primary,
+      var(--tap-sys-color-content-primary)
+    );
+  }
+
+  :host([variant="elevated"]) .button {
+    background-color: var(
+      --tap-button-color-surface-primary,
+      var(--tap-sys-color-surface-primary)
+    );
+    color: var(
+      --tap-button-color-content-primary,
+      var(--tap-sys-color-content-primary)
+    );
+    /* FIXME: we dont have shadow tokens yet  */
+    box-shadow: 0 4px 16px 0 #0000001a;
+  }
+
+  :host([variant="destructive"]) .button {
+    background-color: var(
+      --tap-button-color-surface-destructive,
+      var(--tap-sys-color-surface-negative-light)
+    );
+    color: var(
+      --tap-button-color-content-destructive,
+      var(--tap-sys-color-content-negative)
+    );
+  }
+
+  :host([variant="brand"]) .button {
+    background: var(
+      --tap-button-color-gradient-brand,
+      var(--tap-sys-color-gradient-brand)
+    );
+    color: var(
+      --tap-button-color-content-on-inverse,
+      var(--tap-sys-color-content-on-inverse)
+    );
+  }
+`;
```

Comment:
please only keep the shared styles in the base button styles. and extract the `button` styles to its own style file. for example size related styles.